SabreTools.Serialization/Readers/ISO9660.cs

```diff
@@ -0,0 +1,799 @@
+            if (root)
+                return obj;
+
+            // Calculate actual size of record
+            int totalBytes = 33 + obj.FileIdentifierLength;
+            // Calculate the size of the system use section (remaining allocated bytes)
+            int systemUseLength = obj.DirectoryRecordLength - 33 - obj.FileIdentifierLength;
+            // Account for padding field after file identifier
+            if (obj.FileIdentifierLength % 2 == 0)
+            {
+                totalBytes += 1;
+                systemUseLength -= 1;
+            }
+
+            // If System Use is empty, or if DirectoryRecordLength is bad, return early
+            if (systemUseLength < 1)
+            {
+                // Total record size must be even, read a padding byte
+                if (totalBytes % 2 != 0)
+                    obj.SystemUse = data.ReadBytes(1);
+
+                return obj;
+            }
+
+            // Total used bytes must be even, read a padding byte
+            totalBytes += systemUseLength;
+            if (totalBytes % 2 != 0)
+                systemUseLength += 1;
+
+            // Read system use field
+            obj.SystemUse = data.ReadBytes(systemUseLength);
+
+            return obj;
+        }
+
+        /// <summary>
+        /// Parse a Stream into a DirectoryRecordDateTime
+        /// </summary>
+        /// <param name="data">Stream to parse</param>
+        /// <returns>Filled DirectoryRecordDateTime on success, null on error</returns>
+        public static DirectoryRecordDateTime? ParseDirectoryRecordDateTime(Stream data)
+        {
+            var obj = new DirectoryRecordDateTime();
+
+            obj.YearsSince1990 = data.ReadByteValue();
+            obj.Month = data.ReadByteValue();
+            obj.Day = data.ReadByteValue();
+            obj.Hour = data.ReadByteValue();
+            obj.Minute = data.ReadByteValue();
+            obj.Second = data.ReadByteValue();
+            obj.TimezoneOffset = data.ReadByteValue();
+
+            return obj;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Parse a Stream into a DecDateTime
+        /// </summary>
+        /// <param name="data">Stream to parse</param>
+        /// <returns>Filled DecDateTime on success, null on error</returns>
+        public static DecDateTime ParseDecDateTime(Stream data)
+        {
+            var obj = new DecDateTime();
+
+            obj.Year = data.ReadBytes(4);
+            obj.Month = data.ReadBytes(2);
+            obj.Day = data.ReadBytes(2);
+            obj.Hour = data.ReadBytes(2);
+            obj.Minute = data.ReadBytes(2);
+            obj.Second = data.ReadBytes(2);
+            obj.Centisecond = data.ReadBytes(2);
+            obj.TimezoneOffset = data.ReadByteValue();
+
+            return obj;
+        }
+
+        public static short GetLogicalBlockSize(BaseVolumeDescriptor bvd, short sectorLength)

```

Comment:
This is a good candidate to put in the `Extensions` namespace. It seems like it would be useful outside of just the Reader implementation. If that's not the case, it's fine to leave it here.